grizli/prep.py

```diff
@@ -3146,11 +3180,17 @@ def process_direct_grism_visit(direct={},
         for file in direct['files']:
             crclean = isACS & (len(direct['files']) == 1)
             fresh_flt_file(file, crclean=crclean)
-            try:
-                updatewcs.updatewcs(file, verbose=False, use_db=False)
-            except:
-                updatewcs.updatewcs(file, verbose=False)
+            isJWST = check_isJWST(file)
+            if isJWST:
+                img = jwst_utils.img_with_wcs(file)
+                img.save(file)
+            else:
+                try:
+                    updatewcs.updatewcs(file, verbose=False, use_db=False)
+                except:
```

Comment:
make this `except TypeError:` to explicitly catch "got an unexpected keyword argument"